client/src/actions/newGrant.ts

```diff
@@ -31,33 +27,41 @@ export const grantTXStatus = (status: string): NewGrantActions => ({
   status,
 });
 
-export const grantCreated = ({ id,ipfsHash, owner }: NewGrant): NewGrantActions => ({
+export const grantCreated = ({
+  id,
+  metaData,
+  owner,
+}: NewGrant): NewGrantActions => ({
   type: NEW_GRANT_CREATED,
   id,
-  ipfsHash,
-  owner
+  metaData,
+  owner,
 });
 
-export const mintGrant = () => {
+export const createGrant = () => {
   return async (dispatch: Dispatch, getState: () => RootState) => {
     const state = getState();
     const chainID = state.web3.chainID;
     const addresses = addressesByChainID(chainID!);
-    const signer = global.web3Provider?.getSigner()
-    const grantNFTContract = new ethers.Contract(addresses.grantNft, GrantsRegistryABI, signer)
+    const signer = global.web3Provider?.getSigner();
+    const grantRegistry = new ethers.Contract(
+      addresses.grantsRegistry,
+      GrantsRegistryABI,
+      signer
+    );
     if (state.ipfs.lastFileSavedURL) {
-      const mintTx = await grantNFTContract.mintGrant(state.web3.account, state.ipfs.lastFileSavedURL)
-      dispatch(grantTXStatus('initiated'))
-      const txStatus = await mintTx.wait()
+      const creationTx = await grantRegistry.createGrant(
+        state.web3.account,
+        state.ipfs.lastFileSavedURL,
+        state.web3.account
+      );
+      dispatch(grantTXStatus("initiated"));
+      const txStatus = await creationTx.wait();
       if (txStatus.status) {
-        const grantData = parseMintEvents(txStatus.events)
-        if (!(grantData instanceof Error)) {
-          dispatch(grantTXStatus('complete'))
-          dispatch(grantCreated(grantData))
-        }
+        dispatch(grantTXStatus("complete"));
       } else {
-        throw Error('Unable to mint TX')
+        throw Error("Unable to create grant");
```

Comment:
@schultztimothy we could dispatch the error and manage it, otherwise this would be a runtime error and the user wouldn't have any way to retry